frontend/src/components/filters/Filters.js

```diff
@@ -2,604 +2,131 @@ import counterpart from 'counterpart';
-    const valid = this.isFilterValid(filter);
-
-    this.setState(
-      {
-        notValidFields: !valid,
-      },
-      () => {
-        if (valid) {
-          const parsedFilter = filter.parameters
-            ? {
-                ...filter,
-                parameters: this.parseToPatch(filter.parameters),
-              }
-            : filter;
+   * @param {function} cb - executed if filter is valid and after it was applied
+   */
+  applyFilters = ({ ...filter }, cb) => {
+    const valid = isFilterValid(filter);
+    const { updateNotValidFields, filterId } = this.props;
+
+    updateNotValidFields({ filterId, data: !valid });
+    if (valid) {
+      const parsedFilter = filter.parameters
+        ? {
+            ...filter,
+            parameters: parseToPatch(filter.parameters),
+          }
+        : filter;
 
-          this.setFilterActive(parsedFilter);
+      this.setFilterActive(parsedFilter);
 
-          cb && cb();
-        }
-      }
-    );
+      cb && cb();
+    }
   };
 
   /**
    * @method setFilterActive
-   * @summary This function merges new filters that are to be activated with the existing
-   *  active filters. Additionally we format date fields accordingly so that the backend
-   *  accepts them.
+   * @summary This function updates the active filters we set and then triggers the pre-existing
+   *          logic from DocList that will fetch the filtered data
    * @param {object} filterToAdd
    */
   setFilterActive = (filterToAdd) => {
-    const { updateDocList } = this.props;
-    let { filtersActive } = this.props;
-    const { flatFiltersMap } = this.state;
-    let activeFilters = iMap(filtersActive);
-
-    activeFilters = activeFilters.filter(
-      (item, id) => id !== filterToAdd.filterId
-    );
-    activeFilters = activeFilters.set(filterToAdd.filterId, filterToAdd);
-
-    if (flatFiltersMap) {
-      activeFilters = activeFilters.map((filter, filterId) => {
-        filter.parameters &&
-          filter.parameters.forEach((parameter) => {
-            const { value, valueTo, parameterName } = parameter;
-            const singleFilter = flatFiltersMap[`${filterId}-${parameterName}`];
-
-            if (
-              singleFilter &&
-              DATE_FIELDS.indexOf(singleFilter.widgetType) > -1
-            ) {
-              const format = getFormatForDateField(singleFilter.widgetType);
-
-              if (value) {
-                parameter.value = generateMomentObj(value, format);
-              }
-              if (valueTo) {
-                parameter.valueTo = generateMomentObj(valueTo, format);
-              }
-            }
-          });
+    const { updateDocList, filterId, updateActiveFilter } = this.props;
+    const { filtersActive: storeActiveFilters } = this.props.filters;
 
-        return filter;
-      });
-    }
+    // updating the active filters from the redux store with the filter passed as param
+    const newFiltersActive = setNewFiltersActive({
```

Comment:
is there a case, where we set new active filter without updating active filter ? Maybe `setNewFiltersActive` should call `updateActiveFilter` ?

Reply:
we already have updateActiveFilter (main action that updates the active filter in the redux store with the newFiltersActive that we have set with the newFiltesActive value)